Bypassing security is shoot on sight

RenechCDDA opened the pull request:
#### Summary
Balance "Bypassing faction security is shoot on sight"

#### Purpose of change
Recent conversations pointed out to me that we allow players to bypass faction security and *get away with it* with a warning.

There is no need for a warning. The presence of security measures **is** plenty of warning .

#### Describe the solution
Certain examine actions are no longer minor offenses - jumping fences in particular.

There will be no second chances if you insist on being obviously hostile.

#### Describe alternatives you've considered


#### Testing
Made myself hostile to the artisans with the debug menu, the turrets immediately lit me up instead of just ignoring me.

#### Additional context
I also made the artisan turrets glow, so even if you stumble in in the middle of the night it's obvious that they're there.

Note: The artisans start at +10 `likes_u` so the default -20 when performing a serious offense to them puts them at exactly -10. Just above the threshold for guaranteed hostility.

This is irregular, and I'm not sure why they start at +10.

RenechCDDA committed:
Bypassing security is shoot on sight

PatrikLundell commented:
Will you also get killed if you try to comply with the Tacoma request to place things outside for them to bring in? Doing that required opening the gate, as they won't act on goods they can't see. Before this PR you have to sacrifice standing with them to do so, which is stupid, but at least not lethal.

anoobindisguise commented:
Balthazar is supposed to let you into the intercom room, despite the turrets and bots, right?
That creates a weird thing where if you jump the fence instead of using the gate you get shot even though you're allowed to be there.
Obviously accessing the interior of the bunker should result in you being riddled with new holes, but I recall I was warned about faction ownership when I tried to jump the fence to the compound so I heeded that and took the main gate (no warning given = ok?)

RenechCDDA committed:
Associate monfaction with (NPC) faction, if one exists

Co-authored-by: ehughsbaird <44244083+ehughsbaird@users.noreply.github.com>

RenechCDDA committed:
Stop constructing all factions with "human" as the default mon_faction (???)

RenechCDDA committed:
Do some plumbing for the faction turnover
Put the magic -10 number in one place

RenechCDDA committed:
Retarget https://github.com/CleverRaven/Cataclysm-DDA/pull/62346 at the correct code

Instead of aborting gun target selection after selecting a target, just don't select friendly/neutral characters in the first place as valid targets

RenechCDDA committed:
Actually make monsters hostile when the associated faction will shoot you on sight

(Also add isolated_artisans monster faction to the faction's definition)
